q2_feature_table/tests/test_summarize.py

```diff
@@ -210,10 +212,149 @@ def test_optional_inputs(self):
                 self.assertTrue('<td>3</td>' in file_text)
                 self.assertTrue('<td>4</td>' in file_text)
                 self.assertTrue('<td>6</td>' in file_text)
-                self.assertTrue('<td>1.0</td>' in file_text)
+                self.assertTrue('<td>a;b;e;d</td>' in file_text)
                 self.assertTrue('<td>a;b;c;d</td>' in file_text)
-                self.assertTrue('<td>0.4</td>' in file_text)
-                self.assertTrue('<td>0.7</td>' in file_text)
+                self.assertTrue('<td>08</td>' in file_text)
+                self.assertTrue('<td>17</td>' in file_text)
+
+    def test_optional_input_union(self):
+        seqs = DNAIterator(skbio.DNA(a, metadata=b)for a, b in (
+            ('A', {'id': 'seq01'}),
+            ('AA', {'id': 'seq02'}),
+            ('AAA', {'id': 'seq03'}),
+            ('AAAA', {'id': 'seq04'}),
+            ('AAAA', {'id': 'seq05'}),
+            ('AAA', {'id': 'seq06'}),
+            ('AA', {'id': 'seq07'}),
+            ('AAAAAAAAAA', {'id': 'seq08'})))
+
+        metadata = pd.DataFrame(index=['seq01', 'seq02',
+                                       'seq15', 'seq04',
+                                       'seq05', 'seq90',
+                                       'seq48', 'seq08'],
+                                columns=['att1', 'att2'],
+                                data=[['00', '01'], ['10', '11'],
+                                      ['03', '04'], ['12', '13'],
+                                      ['05', '06'], ['14', '15'],
+                                      ['07', '08'], ['16', '17']])
+        metadata.index.name = 'feature id'
+
+        taxonomy = pd.DataFrame([('a;b;c;d', '1.0'), ('a;b;c;f', '0.7'),
+                                 ('a;b;h;d', '0.3'), ('a;b;d;f', '0.7'),
+                                 ('a;b;e;d', '0.4'), ('a;b;c;f', '0.6'),
+                                 ('a;b;t;d', '1.0'), ('a;b;d;f', '0.5')],
+                                index=['seq17', 'seq02', 'seq03', 'seq48',
+                                       'seq05', 'seq19', 'seq07', 'seq08'],
+                                columns=['Taxon', 'Confidence'])
+
+        metadata = qiime2.Metadata(metadata)
+        taxonomy = {"Taxon Name": [taxonomy]}
+        taxonomy = pd.DataFrame.from_dict(taxonomy)
```

Comment:
Test data for taxonomy should be a dictionary of dataframes. Something like:

```
taxonomy = {
    'foo': pd.DataFrame([('a;b;c;d', '1.0'), ('a;b;c;f', '0.7'),
                                 ('a;b;h;d', '0.3'), ('a;b;d;f', '0.7')],
                                index=['seq17', 'seq02', 'seq03', 'seq48'],
                                columns=['Taxon', 'Confidence'])
    'bar': pd.DataFrame([('a;b;e;d', '0.4'), ('a;b;c;f', '0.6'),
                                 ('a;b;t;d', '1.0'), ('a;b;d;f', '0.5')],
                                index=['seq17', 'seq02', 'seq03', 'seq48''],
                                columns=['Taxon', 'Confidence'])
}
```

Then just pass in that dictionary and assert that `Taxon foo` and `Taxon bar` are in the final viz. 

For union I didn't put all the IDs in both of them but for strict all the IDs would need to be in each of them obviously. The idea here is that we have n different taxonomies that came from different classifiers and we want to put all of them in the viz together.

Reply:
This is effectively what I have done, but I can add more taxonomy data if you would like. Right now there is only one taxonomy, but you would like me to add more? I can definitely do that.